components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -336,52 +329,24 @@ open class GleanInternalAPI internal constructor () {
      */
     fun handleBackgroundEvent() {
         // Schedule the baseline and event pings
-        sendPingsInternal(listOf(BASELINE_STORE_NAME, "events"))
+        sendPings(listOf(Pings.baseline, Pings.events))
     }
 
     /**
-     * Send a list of pings by name.
-     *
-     * Only custom pings (pings not managed by Glean itself) will be sent by
-     * this function. If the name of a Glean-managed ping is passed in, an
-     * error is logged to logcat.
-     *
-     * While the collection of metrics into pings happens synchronously, the
-     * ping queuing and ping uploading happens asyncronously.
-     * There are no guarantees that this will happen immediately.
-     *
-     * If the ping currently contains no content, it will not be sent.
-     *
-     * @param pingNames List of pings to send.
-     * @return true if any pings had content and were queued for uploading
-     */
-    public fun sendPings(pingNames: List<String>): Boolean {
-        val pingsToSend = pingNames.filter { pingName ->
-            if (BUILTIN_PINGNAMES.contains(pingName)) {
-                logger.error("Attempted to send built-in ping $pingName")
-                false
-            } else {
-                true
-            }
-        }
-
-        return sendPingsInternal(pingsToSend)
-    }
-
-    /**
-     * Send a list of pings by name.
+     * Send a list of pings.
      *
      * While the collection of metrics into pings happens synchronously, the
      * ping queuing and ping uploading happens asyncronously.
      * There are no guarantees that this will happen immediately.
      *
-     *
      * If the ping currently contains no content, it will not be sent.
      *
-     * @param pingNames List of pings to send.
+     * @param pings List of pings to send.
+     * @param includeClientInfoOverride If not-null, overrides the `includeClientInfo`
+     * value specified in the [PingType] objects in `pings`.
      * @return true if any pings had content and were queued for uploading
      */
-    internal fun sendPingsInternal(pingNames: List<String>): Boolean {
+    internal fun sendPings(pings: List<PingType>, includeClientInfoOverride: Boolean? = null): Boolean {
```

Comment:
This includes the optional `includeClientInfoOverride` parameter, since I started the implementation before seeing that @georgf said we could safely ignore that for now.  Let me know if it should be left in or removed for now.

Reply:
Let's remove this for now, we decided to slightly change the `activation` ping, so this requirement doesn't need to be satisfied any more.